zaza/openstack/charm_tests/test_utils.py

```diff
@@ -154,6 +154,44 @@ def setUpClass(cls, application_name=None, model_alias=None):
             model_name=cls.model_name)
         logging.debug('Leader unit is {}'.format(cls.lead_unit))
 
+    def config_current_separate_non_string_type_keys(
+            self, non_string_type_keys, config_keys=None,
+            application_name=None):
+        """Obtain current config and the non-string type config separately.
+
+        If the charm config option is not string, it will not accept being
+        reverted back in "config_change()" method if the current value is None.
+        Therefore, obtain the current config and separate those out, so they
+        can be used for a separate invocation of "config_change()" with
+        reset_to_charm_default set to True.
+
```

Comment:
We should make 100% clear here that things will get only separated out if they are currently None, otherwise they won't. Because now it just says something like "if you you don't separate them and they are None, you will get troubles" but it doesn't say that if they aren't None, they won't be separated. Maybe this could be explicitly stated in `:param non_string_type_keys:`